### ISSUE TITLE
TapReporter fails if no config passed

### ISSUE THREAD

**Issue by trjohnst** [testem/testem#1124]
TapReporter before version 1.15.0 allowed to create an instance without passing any arguments. After bumping to later versions, TapReporter required passing a config with a get function otherwise would fail.

If config is not passed in during instantiation, the following line fails: https://github.com/testem/testem/blob/master/lib/reporters/tap_reporter.js#L8

I suggest adding a null check prior to accessing the config:
```
this.quietLogs = config && !!config.get('tap_quiet_logs');
```

**Comment by rwjblue** [comment]
I believe this is basically the same as https://github.com/testem/testem/issues/1115 (but could be wrong).  Sadly, I haven't had a chance to work on the work around I proposed there...

**Comment by trjohnst** [comment]
Thanks @rwjblue, had a chance to start taking a peek into this today. It does look like these issues are the matching the same use case. 

My understanding is a still a little flakey of how everything runs end to end but the route I suggested would most likely flat out ignore configs.

I've got some tests to force the failing use case. I'm working on understanding testem and implementing your suggestion of special handling for `Function` types. I hope to have a PR up at some point today based off of [this fork](https://github.com/trjohnst/testem).

Going to close this out as a duplicate of #1115.
